### ISSUE TITLE
`Style/AccessorGrouping` safe autocorrect results in invalid code

### ISSUE THREAD

**Issue by a-lavis** [rubocop/rubocop#12692]
Sometimes the `Style/AccessorGrouping` cop, which is marked as a safe autocorrect, corrects to incorrect code.

--------

## Expected behavior

When running `rubocop -a foo.rb` where `foo.rb` is the following file:

```ruby
# frozen_string_literal: true

class Foo # :nodoc:
  attr_reader :one

  OTHER_ATTRS = %i[two three].freeze

  attr_reader(*OTHER_ATTRS)
end
```

I expect `Style/AccessorGrouping` to either not safe autocorrect, or for it to safe autocorrect to:

```ruby
# frozen_string_literal: true

class Foo # :nodoc:
  OTHER_ATTRS = %i[two three].freeze

  attr_reader :one, *OTHER_ATTRS
end
```

## Actual behavior

It safe autocorrects to:

```ruby
# frozen_string_literal: true

class Foo # :nodoc:
  attr_reader :one, *OTHER_ATTRS

  OTHER_ATTRS = %i[two three].freeze
end
```

This code is invalid, because `OTHER_ATTRS` is used before it is defined.

## Steps to reproduce the problem

Run `rubocop -a foo.rb` where `foo.rb` is the above file.

## RuboCop version

```
> rubocop -V
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.3.0) [arm64-darwin23]
```
